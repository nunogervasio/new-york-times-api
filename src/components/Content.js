import React from "react";
import MostViewed from "./MostViewed";
import TopStories from "./TopStories";
import TopStoriesTech from "./TopStoriesTech";
import TopStoriesScience from "./TopStoriesScience";
import { Grid } from "semantic-ui-react";
import TopStoriesArts from "./TopStoriesArts";

const Content = () => {
  return (
    <React.Fragment>
      <Grid centered={true}>
        <Grid.Column
          widescreen={6}
          largeScreen={8}
          computer={7}
          tablet={9}
          mobile={14}
        >
          <MostViewed />
          <TopStoriesArts />
        </Grid.Column>

        <Grid.Column
          widescreen={4}
          largeScreen={6}
          computer={7}
          tablet={5}
          mobile={14}
        >
          <Grid columns="equal">
            <Grid.Column tablet={16} computer={8}>
              <TopStories />
              <TopStoriesScience />
            </Grid.Column>
            <Grid.Column only="computer">
              <TopStoriesTech />
            </Grid.Column>
          </Grid>
        </Grid.Column>
      </Grid>
    </React.Fragment>
  );
};

export default Content;
