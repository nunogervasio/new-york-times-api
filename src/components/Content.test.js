import React from "react";
import { shallow } from "enzyme";
import Content from "./Content";

describe("Content", () => {
  const content = shallow(<Content />);

  it("renders properly", () => {
    expect(content).toMatchSnapshot();
  });
  it("renders MostViewed", () => {});
  it("renders TopStoriesArts", () => {});
  it("renders TopStories", () => {});
  it("renders TopStoriesScience", () => {});
  it("renders TopStoriesTech", () => {});
});
