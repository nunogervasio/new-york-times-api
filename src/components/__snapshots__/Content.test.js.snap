// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Content />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Grid
        centered={true}
      >
        <GridColumn
          computer={7}
          largeScreen={8}
          mobile={14}
          tablet={9}
          widescreen={6}
        >
          <MostViewed />
          <TopStoriesArts />
        </GridColumn>
        <GridColumn
          computer={7}
          largeScreen={6}
          mobile={14}
          tablet={5}
          widescreen={4}
        >
          <Grid
            columns="equal"
          >
            <GridColumn
              computer={8}
              tablet={16}
            >
              <TopStories />
              <TopStoriesScience />
            </GridColumn>
            <GridColumn
              only="computer"
            >
              <TopStories />
            </GridColumn>
          </Grid>
        </GridColumn>
      </Grid>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "centered": true,
        "children": Array [
          <GridColumn
            computer={7}
            largeScreen={8}
            mobile={14}
            tablet={9}
            widescreen={6}
          >
            <MostViewed />
            <TopStoriesArts />
          </GridColumn>,
          <GridColumn
            computer={7}
            largeScreen={6}
            mobile={14}
            tablet={5}
            widescreen={4}
          >
            <Grid
              columns="equal"
            >
              <GridColumn
                computer={8}
                tablet={16}
              >
                <TopStories />
                <TopStoriesScience />
              </GridColumn>
              <GridColumn
                only="computer"
              >
                <TopStories />
              </GridColumn>
            </Grid>
          </GridColumn>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <MostViewed />,
              <TopStoriesArts />,
            ],
            "computer": 7,
            "largeScreen": 8,
            "mobile": 14,
            "tablet": 9,
            "widescreen": 6,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Grid
              columns="equal"
            >
              <GridColumn
                computer={8}
                tablet={16}
              >
                <TopStories />
                <TopStoriesScience />
              </GridColumn>
              <GridColumn
                only="computer"
              >
                <TopStories />
              </GridColumn>
            </Grid>,
            "computer": 7,
            "largeScreen": 6,
            "mobile": 14,
            "tablet": 5,
            "widescreen": 4,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <GridColumn
                  computer={8}
                  tablet={16}
                >
                  <TopStories />
                  <TopStoriesScience />
                </GridColumn>,
                <GridColumn
                  only="computer"
                >
                  <TopStories />
                </GridColumn>,
              ],
              "columns": "equal",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <TopStories />,
                    <TopStoriesScience />,
                  ],
                  "computer": 8,
                  "tablet": 16,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <TopStories />,
                  "only": "computer",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Grid
          centered={true}
        >
          <GridColumn
            computer={7}
            largeScreen={8}
            mobile={14}
            tablet={9}
            widescreen={6}
          >
            <MostViewed />
            <TopStoriesArts />
          </GridColumn>
          <GridColumn
            computer={7}
            largeScreen={6}
            mobile={14}
            tablet={5}
            widescreen={4}
          >
            <Grid
              columns="equal"
            >
              <GridColumn
                computer={8}
                tablet={16}
              >
                <TopStories />
                <TopStoriesScience />
              </GridColumn>
              <GridColumn
                only="computer"
              >
                <TopStories />
              </GridColumn>
            </Grid>
          </GridColumn>
        </Grid>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "centered": true,
          "children": Array [
            <GridColumn
              computer={7}
              largeScreen={8}
              mobile={14}
              tablet={9}
              widescreen={6}
            >
              <MostViewed />
              <TopStoriesArts />
            </GridColumn>,
            <GridColumn
              computer={7}
              largeScreen={6}
              mobile={14}
              tablet={5}
              widescreen={4}
            >
              <Grid
                columns="equal"
              >
                <GridColumn
                  computer={8}
                  tablet={16}
                >
                  <TopStories />
                  <TopStoriesScience />
                </GridColumn>
                <GridColumn
                  only="computer"
                >
                  <TopStories />
                </GridColumn>
              </Grid>
            </GridColumn>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <MostViewed />,
                <TopStoriesArts />,
              ],
              "computer": 7,
              "largeScreen": 8,
              "mobile": 14,
              "tablet": 9,
              "widescreen": 6,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Grid
                columns="equal"
              >
                <GridColumn
                  computer={8}
                  tablet={16}
                >
                  <TopStories />
                  <TopStoriesScience />
                </GridColumn>
                <GridColumn
                  only="computer"
                >
                  <TopStories />
                </GridColumn>
              </Grid>,
              "computer": 7,
              "largeScreen": 6,
              "mobile": 14,
              "tablet": 5,
              "widescreen": 4,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <GridColumn
                    computer={8}
                    tablet={16}
                  >
                    <TopStories />
                    <TopStoriesScience />
                  </GridColumn>,
                  <GridColumn
                    only="computer"
                  >
                    <TopStories />
                  </GridColumn>,
                ],
                "columns": "equal",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [
                      <TopStories />,
                      <TopStoriesScience />,
                    ],
                    "computer": 8,
                    "tablet": 16,
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <TopStories />,
                    "only": "computer",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": true,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
